allensdk/brain_observatory/behavior/eye_tracking_processing.py

```diff
@@ -0,0 +1,164 @@
+from pathlib import Path
+
+import numpy as np
+import pandas as pd
+
+from scipy import ndimage, stats
+
+
+def load_eye_tracking_hdf(eye_tracking_file: Path) -> pd.DataFrame:
+    """Load a DeepLabCut hdf5 file containing eye tracking data into a
+    dataframe.
+
+    Note: The eye tracking hdf5 file contains 3 separate dataframes. One for
+    corneal reflection (cr), eye, and pupil ellipse fits. This function
+    loads and returns this data as a single dataframe.
+
+    Parameters
+    ----------
+    eye_tracking_file : Path
+        Path to an hdf5 file produced by the DeepLabCut eye tracking pipeline.
+        The hdf5 file will contain the following keys: "cr", "eye", "pupil".
+        Each key has an associated dataframe with the following
+        columns: "center_x", "center_y", "height", "width", "phi".
+
+    Returns
+    -------
+    pd.DataFrame
+        A dataframe containing combined corneal reflection (cr), eyelid (eye),
+        and pupil data. Column names for each field will be renamed by
+        prepending the field name. (e.g. center_x -> eye_center_x)
+    """
+    eye_tracking_fields = ["cr", "eye", "pupil"]
+
+    eye_tracking_dfs = []
+    for field_name in eye_tracking_fields:
+        field_data = pd.read_hdf(eye_tracking_file, key=field_name)
+        new_col_name_map = {col_name: f"{field_name}_{col_name}"
+                            for col_name in field_data.columns}
+        field_data.rename(new_col_name_map, axis=1, inplace=True)
+        eye_tracking_dfs.append(field_data)
+
+    eye_tracking_data = pd.concat(eye_tracking_dfs, axis=1)
+    eye_tracking_data.index.name = 'frame'
+
+    # Values in the hdf5 may be complex (likely an artifact of the ellipse
+    # fitting process). Take only the real component.
+    eye_tracking_data = eye_tracking_data.apply(lambda x: np.real(x.to_numpy()))
+
+    return eye_tracking_data.astype(float)
+
+
+def determine_outliers(eye_tracking_df: pd.DataFrame,
```

Comment:
I don't completely understand how this will work. Will it check *every* column for outliers and return TRUE if *any* value exceeds a z-score of `z_threshold`? 

**I think it should only operate on the area columns**. In the equivalent VBA implementation, I discovered that transient eye movements can be sufficiently large and rare such that they have a large z-score, but they are still perfectly captured by the ellipse fits. Also, since the `outliers` column is used in the `determine_likely_blinks` method, these large ballistic eye movements end up being labeled as likely_blink == True, which is not correct.

Reply:
This currently operates on *all* columns. As my thinking was that extremely large z-scores in any of the parameters would suggest something had gone wrong with the tracking. Given that you've looked and seen that ellipse fits are still accurate, I can change this to only operate on `eye_area` and `pupil_area` columns.